primary/src/helper.rs

```diff
@@ -1,41 +1,50 @@
 // Copyright (c) 2021, Facebook, Inc. and its affiliates
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-use crate::primary::PrimaryMessage;
-use config::Committee;
+use crate::{primary::PrimaryMessage, PayloadToken};
+use config::{Committee, WorkerId};
 use crypto::traits::VerifyingKey;
 use network::PrimaryNetwork;
 use store::Store;
 use tokio::sync::mpsc::Receiver;
 use tracing::{error, warn};
-use types::{Certificate, CertificateDigest};
+use types::{BatchDigest, Certificate, CertificateDigest};
 
 #[cfg(test)]
 #[path = "tests/helper_tests.rs"]
 mod helper_tests;
 
-/// A task dedicated to help other authorities by replying to their certificates requests.
+/// A task dedicated to help other authorities by replying to their certificate &
+/// payload availability requests.
 pub struct Helper<PublicKey: VerifyingKey> {
+    /// The node's name
+    name: PublicKey,
     /// The committee information.
     committee: Committee<PublicKey>,
-    /// The persistent storage.
-    store: Store<CertificateDigest, Certificate<PublicKey>>,
-    /// Input channel to receive certificates requests.
+    /// The certificate persistent storage.
+    certificate_store: Store<CertificateDigest, Certificate<PublicKey>>,
+    /// The payloads (batches) persistent storage.
+    payload_store: Store<(BatchDigest, WorkerId), PayloadToken>,
```

Comment:
Nit : I wonder if we're not creating confusion by introducing the keyword `Payload` in many names, where "BatchData" would have sufficed using old concepts.

Reply:
Ah, that's a very good point @huitseeker and I think we might need to move into some renaming in general. I did stick to the `payload_` naming convention that has been used in several places (especially around the store semantics). I would be happy to align in one name (e.x `batch`) and stick with that everywhere

Reply:
Here's my plug to ruplacer: https://github.com/dmerejkowsky/ruplacer

Reply:
oh nice, thanks!